server/api/bots.js

```diff
@@ -2,29 +2,30 @@ const express = require("express");
 const router = express.Router();
 const Bots = require("../database/models/Bot.js");
 const Users = require("../database/models/User");
+const botApproveMethods = require("../constants/botApproveMethods");
+const { getBotData } = require("../util/getBotData");
 
-//Need a good error codes for all of these. Please help
-
+// get all bots from db
 router.get("/", async (req, res) => {
-    // get all bots from db
     const allBots = await Bots.find();
     return res.json(allBots);
 });
-
+//gets one bot based on its id
 router.get("/:id", async (req, res) => {
     const { id } = req.params;
     const foundBot = await Bots.findOne({ id });
     if (!foundBot) return res.status(404).json({ message: "A bot was not found!", error: "Not Found." });
     return res.json(foundBot);
 });
-
+//posts a bot to the db
 router.post("/", async (req, res) => {
-    const { id, name, prefix, description, owners, website, helpCommand, supportServer, library } = req.body;
-    if (!id || !name || !prefix || !description || !owners || !website || !helpCommand || !supportServer || !library) return res.status(404).json({ msg: "Your missing some information to create the bot!" });
+    const { id, prefix, description, owners, website, helpCommand, supportServer, library } = req.body;
+    if (!id || !prefix || !description || !owners || !website || !helpCommand || !supportServer || !library) return res.status(404).json({ msg: "Your missing some information to create the bot!" });
     const bot = await Bots.findOne({ id });
-    //need a good error code
+    const botApiData = await getBotData(id);
+    const { tag, avatarUrl } = botApiData;
     if (bot) return res.status(400).send({ message: "This bot already exists!", error: "Bad Request." });
-    const newBot = new Bots({ id, name, prefix, description, owners, website, helpCommand, supportServer, library });
+    const newBot = new Bots({ id, tag, avatarUrl, prefix, description, owners, website, helpCommand, supportServer, library });
```

Comment:
cuz im getting the full tag from here. no need for name anymore

Reply:
Ok